dask_sql/physical/rex/core/call.py

```diff
@@ -634,6 +634,132 @@ def timestampadd(self, unit, interval, df: SeriesOrScalar):
             raise NotImplementedError(f"Extraction of {unit} is not (yet) implemented.")
 
 
+class DatetimeSubOperation(Operation):
+    """
+    Datetime subtraction is a special case of the `minus` operation
+    which also specifies a sql interval return type for the operation.
+    """
+
+    def __init__(self):
+        super().__init__(self.datetime_sub)
+
+    def datetime_sub(self, unit, df1, df2):
+        subtraction_op = ReduceOperation(
+            operation=operator.sub, unary_operation=lambda x: -x
+        )
+        result = subtraction_op(df2, df1)
+
+        if unit in {"NANOSECOND", "NANOSECONDS"}:
+            return result
+        elif unit in {"MICROSECOND", "MICROSECONDS"}:
+            return result // 1_000
+        elif unit in {"SECOND", "SECONDS"}:
+            return result // 1_000_000_000
+        elif unit in {"MINUTE", "MINUTES"}:
+            return (result / 1_000_000_000) // 60
+        elif unit in {"HOUR", "HOURS"}:
+            return (result / 1_000_000_000) // 3600
+        elif unit in {"DAY", "DAYS"}:
+            return ((result / 1_000_000_000) / 3600) // 24
+        elif unit in {"WEEK", "WEEKS"}:
+            return (((result / 1_000_000_000) / 3600) / 24) // 7
+        elif unit in {"MONTH", "MONTHS"}:
+            # TODO: Figure out a way to do this without calling compute()
+            result = []
+            for ind in range(len(df1)):
+                year1 = df1[ind].dt.year.compute()[ind]
+                year2 = df2[ind].dt.year.compute()[ind]
+                diff = (year2 - year1) * 12
+
+                month1 = df1[ind].dt.month.compute()[ind]
+                month2 = df2[ind].dt.month.compute()[ind]
+                diff += month2 - month1
+
+                day1 = df1[ind].dt.day.compute()[ind]
+                day2 = df2[ind].dt.day.compute()[ind]
+
+                if diff > 0 and day1 > day2:
+                    diff = diff - 1
+                elif diff < 0 and day2 > day1:
+                    diff = diff + 1
+
+                result.append(diff)
```

Comment:
Trying to think of a way we can do this without compute. 
Is it possible to somehow divide the result by `np.timedelta64(1, "M")`.

Happy to discuss offline if this approach works

Reply:
Thanks! Initially, I was having rounding issues with months and quarters, (e.g., just doing # of days / 30 was not enough to produce an exactly correct result), but I've updated my implementation to divide by a more precise number (e.g., the average number of days in a month is `((30 * 4) + 28 + (31 * 7)) / 12`), which has removed the need for `compute()`.

For years, I initially had extra logic to deal with leap days, particularly the cases when we are comparing February 28/29 and/or March 1, which required `compute()` to extract the months and days to compare with each other. This is pretty nitpicky, though, so I removed that logic and just calculate the years with # of days / 365.

Let me know if the logic for these changes seems reasonable 🙂

Reply:
The changes make sense to me. Even libraries like numpy/mysql have differing behavior here and diff between months/leap years are handled differently in many cases.